### PR TITLE
Pick language from page translation over page context if available

### PR DESCRIPTION
**Pull request by wobba**

### CODE REVIEW

File: search-parts/src/dataSources/SharePointSearchDataSource.ts [hunk 1]
```diff
@@ -190,6 +191,15 @@ export class SharePointSearchDataSource extends BaseDataSource<ISharePointSearch
         }
     }
 
+    private getTranslatedCultureFromUrl(): string {
```
Comment (FranckyC) [inline search-parts/src/dataSources/SharePointSearchDataSource.ts:194]: Any example strings that match this exp? From what I see it, any page within a folder will match this resulting to an incorrect language (ex: "/SitePages/test/test.aspx"). The regex should be more precise to match a culture format.

Comment (wobba) [reply search-parts/src/dataSources/SharePointSearchDataSource.ts:194]: Good comment, but for modern you don't have pages in Folders... unless you hack it in. So still a concern?

Comment (wobba) [reply search-parts/src/dataSources/SharePointSearchDataSource.ts:203]: You can make folders, but not make modern pages inside them. Committed fallback if culture resolves to 0.

Comment (FranckyC) [reply search-parts/src/dataSources/SharePointSearchDataSource.ts:203]: image

It seems you can :D

Comment (wobba) [reply search-parts/src/dataSources/SharePointSearchDataSource.ts:203]: They keep changing things :D I don't have "create folder" as an option in my SitePage lib as default. Either way, the fallback should work just fine. It's not intended to create folders 😉

Comment (FranckyC) [reply search-parts/src/dataSources/SharePointSearchDataSource.ts:203]: > 
> 
> They keep changing things :D I don't have "create folder" as an option in my SitePage lib as default. Either way, the fallback should work just fine. It's not intended to create folders 😉

Folders are used for security reasons on navigation links, that is the only reason imo. Anyway, just consider this scenario as well in the regex

Comment (wobba) [reply search-parts/src/dataSources/SharePointSearchDataSource.ts:203]: Changed the logic to take the last folder name as culture candidate. Tested and verified with subfolders and translations 👍

**Review by FranckyC — APPROVED** [review]

**Pull request merged**